From 4af686967e03ce24482d1510887140ab1da8df45 Mon Sep 17 00:00:00 2001
From: fatalc <cnfatal@gmail.com>
Date: Tue, 17 Feb 2026 03:15:36 +0800
Subject: [PATCH 7/8] Pin pyobjus to 694ce31 and use Cython build_ext for iOS

Pin to commit 694ce31 (last Python 2 compatible commit).
The iOS code path in setup.py expects a pre-generated pyobjus.c,
but this file was never shipped in the git repo. Patch setup.py
to use Cython.Distutils build_ext on iOS (same as macOS path).
Make Cython available to hostpython via PY2_VENV PYTHONPATH.
---
 dependencies/scripts/build-pyobjus.sh | 16 +++++++++++++++-
 1 file changed, 15 insertions(+), 1 deletion(-)

diff --git a/dependencies/scripts/build-pyobjus.sh b/dependencies/scripts/build-pyobjus.sh
index 0731f90..3756e6c 100755
--- a/dependencies/scripts/build-pyobjus.sh
+++ b/dependencies/scripts/build-pyobjus.sh
@@ -4,12 +4,21 @@
 
 
 
+# Pin to 694ce31 - last Python 2 compatible commit (supports Python 2.7/3.5-3.7)
+PYOBJUS_COMMIT=694ce31
+
 if [ ! -d $TMPROOT/pyobjus ] ; then
-    git clone https://github.com/kivy/pyobjus.git $TMPROOT/pyobjus
+    try git clone https://github.com/kivy/pyobjus.git $TMPROOT/pyobjus && git -C $TMPROOT/pyobjus checkout -f $PYOBJUS_COMMIT
 fi
 
 try pushd $TMPROOT/pyobjus
 
+# The iOS path in setup.py expects a pre-generated pyobjus.c, but this file was
+# never shipped in the git repo. Patch setup.py to use Cython on iOS just like
+# it does on macOS - this is the standard build_ext mechanism, not preprocessing.
+sed -i '' "s/from distutils.command.build_ext import build_ext/from Cython.Distutils import build_ext/" setup.py
+sed -i '' "s/files = \['pyobjus.c'\]/files = ['pyobjus.pyx']/" setup.py
+
 # Set environment variables for Python module cross-compile
 OLD_CC="$CC"
 OLD_CFLAGS="$CFLAGS"
@@ -21,6 +30,11 @@ export LDSHARED="$RENIOSDEPROOT/scripts/liblink"
 
 HOSTPYTHON="$RENIOSDEPROOT/tmp/Python-$PYTHON_VERSION/hostpython"
 
+# Make Cython available to hostpython via PY2_VENV site-packages
+if [ -n "$PY2_VENV" ]; then
+    export PYTHONPATH="$PY2_VENV/lib/python2.7/site-packages${PYTHONPATH:+:$PYTHONPATH}"
+fi
+
 export KIVYIOSROOT=1
 export ARCH=$RENIOSARCH
 
-- 
2.52.0

